pkg/codegen/testing/test/testdata/azure-native-nested-types/nodejs/types/input.ts

```diff
@@ -4,6 +4,6 @@
 import * as pulumi from "@pulumi/pulumi";
 import * as inputs from "../types/input";
 import * as outputs from "../types/output";
+import * as utilities from "../utilities";
 
-export namespace documentdb {
-}
+export * as documentdb from "./documentdb/input";
```

Comment:
Hm, interesting! So it only re-exports /input.

Reply:
Yeah, that was what I was getting hung up on for the longest time.
I tried:

```typescript
export * as documentdb from "./documentdb";
```

which would export everything in the index file, but that had some knock-on effects to preserve the API:

1. If I tried to have the index file re-export both inputs and outputs in the _same_ namespace, this would introduce a name collision between `documentdb/inputs.ts::Foo: Input<T>` and `documentdb/outputs.ts::Foo: Output<T>` (or something to this effect).
2. If I tried to have the index file re-export `inputs.ts as inputs` and `outputs.ts as outputs`, then I would break backward-compat. You'd have to import things with this syntax `import "types/inputs/documentdb/inputs/..."`.

This approach agglomerates the nested input items in the parent's namespace, and same with outputs. This preserves backward-compat with what we have today.

Reply:
This Is the key in the PR. Brilliant.